File: src/main/java/atm/bloodworkxgaming/serverstarter/config/LockFile.java
package atm.bloodworkxgaming.serverstarter.config;

import lombok.EqualsAndHashCode;
import lombok.ToString;

import java.util.Objects;

@EqualsAndHashCode
@ToString
public class LockFile {
    public boolean forgeInstalled = false;
    public boolean packInstalled = false;
    public String forgeVersion = "NONE";
    public String mcVersion = "NONE";
    public String packUrl = "NONE";
    public String spongeBootstrapper = "NONE";


    public boolean checkShouldInstall(ConfigFile configFile) {
        return !forgeInstalled
                || !packInstalled
                || (configFile.install.forgeVersion != null && !configFile.install.forgeVersion.isEmpty() && !Objects.equals(forgeVersion, configFile.install.forgeVersion))
                || !Objects.equals(packUrl, configFile.install.modpackUrl);
    }
}

File: src/main/kotlin/atm/bloodworkxgaming/serverstarter/config/LockFile.kt
package atm.bloodworkxgaming.serverstarter.config

data class LockFile(
        var loaderInstalled: Boolean = false,
        var packInstalled: Boolean = false,
        var loaderVersion: String = "NONE",
        var mcVersion: String = "NONE",
        var packUrl: String = "NONE",
        var spongeBootstrapper: String = "NONE") {


    fun checkShouldInstall(configFile: ConfigFile): Boolean {
        return (!loaderInstalled
                || !packInstalled
                || !configFile.install.loaderVersion.isNullOrEmpty() && loaderVersion != configFile.install.loaderVersion
                || packUrl != configFile.install.modpackUrl)
    }
}
